templates/service/app/views/Layout.scala.html

```diff
@@ -1,7 +1,14 @@
 @import uk.gov.hmrc.$!APP_PACKAGE_NAME!$.config.AppConfig
+@import uk.gov.hmrc.govukfrontend.views.html.components.GovukLayout
+@import uk.gov.hmrc.hmrcfrontend.views.html.helpers.HmrcStandardFooter
```

Comment:
While I agree with removing the TwirlKeys.templateImports and leave this up to developer choice, my opinion is the imports should as suggested in the play-frontend-govuk and play-frontend-hmrc documentation as:

 ```
 @import uk.gov.hmrc.govukfrontend.views.html.components._
 @import uk.gov.hmrcfrontend.views.html.helpers._
 ```
 
The rationale is that the frontend library package objects bring in a lot of implicit classes and helpers and case classes that make using the library much easier for Scala developers. Not bringing these in automatically is likely to make life painful for teams implementing these libraries.

The dependency is also then with the package rather than individual classes (looser coupling)

I'm with Robert C. Martin on this one :-)

Reply:
I would personally avoid wildcard imports apart from importing implicits. It avoids ambiguity (there are collisions in the hmrcfrontend and govukfrontend packages), and makes reading code without relying on an IDE easier (e.g. PRs).

The only decoupling I can see is that the class/function you are using could move between imported packages. But I'd consider the class and the package it resides in as the public API.